src/tiled/stampbrush.cpp

```diff
@@ -385,7 +388,53 @@ void StampBrush::drawPreviewLayer(const QVector<QPoint> &list)
             const TileStampVariation variation = mStamp.randomVariation();
             mapDocument()->unifyTilesets(variation.map, mMissingTilesets);
 
-            TileLayer *stamp = variation.tileLayer();
+            TileLayer *stamp = variation.tileLayer()->copy(variation.tileLayer()->bounds());
+
+            //If using a hexagonal map, this takes into acount staggering
+            if(mapDocument()->map()->orientation() == 4 && stamp->height()>1)
+            {
+                int shiftMode;//-1 do not shift, 0 shift even rows (relatived to the stamp), 1 shift odd rows
+                int rowOfTop = p.y()-stamp->height()/2;
+                if(rowOfTop<0)rowOfTop*=-1; //Only used to check if even or odd, and being negative can mess this up.
```

Comment:
Always add spaces to make the code more readable. Also, add the statement `rowOfTop*=-1` in the next line :)
```
if (rowOfTop < 0)
    rowOfTop *= -1;